### PR TITLE
Push docker images betterer

### PR DESCRIPTION
**Pull request by meatballhat**
tl;dr - the content of `post-processors` may seem confusing, but it's for good-ish reasons

Each entry in the top-level `post-processors` array is effectively a mini pipeline, where the result(s) of a given post-processor are available to subsequent post-processors.  What's more, certain post-processors such as `shell-local` will be run once for each *file* artifact, meaning that if there are no *file* artifacts, they will not run at all.  This is one of the reasons why there is an `artifice` post-processor that explicitly sets the `files` array to `tmp/docker-meta/.dumped`.  The other reasons why this `artifice` post-processor is doing this is because the `shell-local` post-processor logs its entire command line with env vars, which would mean exposing secrets in public logs.  By "dumping" the docker-specific env vars (including credentials) into `tmp/docker-meta` as an envdir, then passing a sentinel file of `tmp/docker-meta/.dumped` to the `shell-local` file as a fake artifact, we can then ensure that the `shell-local` post-processor runs if and only if the docker secret env vars are available.

### CODE REVIEW

**Comment by meatballhat** [comment]
Looking classy, quay (sample size: 1)

**Comment by meatballhat** [comment]
yaaaas looking *good*
